wolox-books/src/app/screens/user/components/login/login.component.ts

```diff
@@ -7,9 +10,35 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
-  constructor() { }
+  formGroup: FormGroup;
+
+  constructor(private readonly formBuilder: FormBuilder, private userService: UserService) {
+    this.formGroup = formBuilder.group({
+      'email': [null, [Validators.required, Validators.email]],
+      'password': [null, [Validators.required, Validators.pattern('(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}')]]
+    });
+  }
+
+  get email() {
+    return this.formGroup.controls.email;
+  }
+
+  get password() {
+    return this.formGroup.controls.password;
+  }
 
   ngOnInit(): void {
   }
 
+  login() {
```

Comment:
I think form values are sent as params here

Reply:
🤔 I passed variables as parameters but @betoobernal said me that It was better do it like this, because if the variable exists on the class, why do we send it as parameters?

Reply:
Yes keep it as parameters